사장님 가게 목록 조회

TIP-Santa opened the pull request:
#### 시나리오

---

사장님 가게 목록 조회
가게 이름, 가게 영업 상태 조회
가게 상태가 폐업인 경우 조회되지 않음

#### 가게 조회 테스트
---

- 가게 조회 성공
- 가게 조회 실패 케이스
  - 해당 owner Id에 등록된 store가 없는 경우

TIP-Santa committed:
feat : 가게 조회 기능 구현 #30
- ErrorCode : STORE_NOT_FOUND 추가
- GetStoreRespDto 추가
- StoreController : 가게 조회 기능 추가
- StoreService : 가게 조회 기능 추가
- StoreRepository : findByOwnerId 추가

TIP-Santa committed:
refactor : 가게 조회 #30
- GetStoreRespDto : 목록에서 보여질 데이터는 이름과 영업상태만으로 수정

TIP-Santa committed:
refactor : 가게 조회
- GetStoreRespDto : 코드 정리

TIP-Santa committed:
refactor : 가게 조회 #30
- Store : isDeleted 초기값 설정
- StoreController : 코드 포멧팅 정리
- StoreCreateReqDto : store 생성 시 isDeleted값은 초기값 false로 설정관련 isDeleted 필드 제거
- StoreService : 폐업상태인 가게는 조회에서 제외 처리

TIP-Santa committed:
test : 가게 조회 #30
- StoreServiceTest
  - 성공 테스트 : 영업상태가 폐업인 가게 조회 제외
  - 실패 테스트 : 해당 ownerId에 생성된 Store가 없는 경우

TIP-Santa committed:
dev pull > merge

TIP-Santa committed:
merge

JIMINJI1 reviewed (approved):
확인했습니다

yaewon0411 commented on src/main/java/com/my/foody/domain/store/controller/StoreController.java:
```diff
@@ -23,8 +26,16 @@ public class StoreController {
     }
+
+    @RequireAuth(userType = UserType.OWNER)
+    @GetMapping("stores")
```
/가 누락되었습니다

TIP-Santa replied:
앗 감사합니다! 잔실수가 많네요;;

yaewon0411 commented on src/main/java/com/my/foody/domain/store/service/StoreService.java:
```diff
@@ -45,16 +46,19 @@ public StoreCreateRespDto createStore(StoreCreateReqDto storeCreateReqDto, Long
+                .filter(store -> !store.getIsDeleted())
+                .toList();
+        // 해당 ID의 가게가 없는 경우
+        if (storeList.isEmpty()) {
+            throw new BusinessException(ErrorCode.STORE_NOT_FOUND);
         }
```
owner의 가게가 없는 건 예외 상황이 아니라고 생각합니다
이 경우에도 빈 리스트를 내보내야 합니다

TIP-Santa replied:
아 그렇겠네요 감사합니다!
예외처리 삭제했습니다.
```
public List<GetStoreRespDto> getAllStoresByOwnerId(Long ownerId) {
        // 해당 ID의 가게 조회 (영업중인 가게만 조회)
        List<Store> storeList = storeRepository.findByOwnerId(ownerId)
                .stream()
                .filter(store -> !store.getIsDeleted())
                .toList();
        return storeList.stream()
                .map(store -> new GetStoreRespDto(store))
                .toList();
    }
```

yaewon0411 commented on src/main/java/com/my/foody/domain/store/service/StoreService.java:
```diff
@@ -45,16 +46,19 @@ public StoreCreateRespDto createStore(StoreCreateReqDto storeCreateReqDto, Long
+        if (storeList.isEmpty()) {
+            throw new BusinessException(ErrorCode.STORE_NOT_FOUND);
         }
+        return storeList.stream()
+                .map(store -> new GetStoreRespDto(store))
+                .toList();
```
GetStoreRespDto를 보면 isDeleted필드를 내보내고 있는데
위에서 이미 filter를 통해 store가 isDelete=false인 것만을 내보내고 있어서 모든 isDeleted는 false값만을 갖게 됩니다

저는 위에서 filter없이 모든 store를 내보내고, owner입장에서도 본인이 폐업 상태로 전환한 store를 볼 수 있어야 한다고 생각합니다 그래야 나중에 영업 전환으로 다시 변환한다 등의 기능이 가능할 것으로 보여요

TIP-Santa replied:
사실 저도 이 부분이 햇갈리고있습니다.
저희 프로잭트 조건에서는 가게 폐업을 하게 되면 가게 조회 시 나타나지 않는다는 조건이 있어서 이게 사장님이 보는 가게 목록에도 해당되는건지가 조금 햇갈렸습니다.
그럼 사장님이 볼 때는 폐업처리된 가게도 보이도록 수정하겠습니다.

TIP-Santa replied:
가게 조회 시 영업상태 조회조건 제거했습니다.
```
public List<GetStoreRespDto> getAllStoresByOwnerId(Long ownerId) {
        // 해당 ID의 가게 조회
        List<Store> storeList = storeRepository.findByOwnerId(ownerId);

        return storeList.stream()
                .map(store -> new GetStoreRespDto(store))
                .toList();
    }
```

TIP-Santa committed:
refactor 리뷰 반영 #30
- ErrorCode : STORE_NOT_FOUND 제거
- Store : isDeleted 설정값 제거
- StoreController : GetMapping에 빠진 / 추가
- StoreService : 폐업중인 가게도 조회하기 위해 조회조건 제거
- StoreRepository : 가게 생성 시 영업중인 가게를 카운트하기 위해 검색 조건 추가
- StoreCreateReqDto : 가개 생성 시 isDeleted값 false로 기본 설정
- StoreCreateRespDto : 가게 생성 성공 시 메세지 출력 관련 필요없는 필드 제거
- Store : RepDto에서 isDeleted 기본값 설정 관련 해당 클래스에서는 기본값 설정 제거
- StoreServiceTest : 가게가 없는 경우는 실패가 아니기에 제거

TIP-Santa committed:
refactor 리뷰 반영 #30
- StoreService : 주석 수정

TIP-Santa committed:
dev full > merge

yaewon0411 commented:
```
@Getter
public class GetStoreRespDto {
    private String name;
    private boolean isDeleted;

    public GetStoreRespDto(Store store) {
        this.name = store.getName();
        this.isDeleted = store.getIsDeleted();
    }
}
```
이미 생성자를 구현한 상태에서는 @NoArgs가 안달려있으면 Json 직렬화가 안됩니다ㅠ

아마 컨트롤러 단위 테스트를 해보시면 오류가 날 겁니다

yaewon0411 commented:
```
    @RequireAuth(userType = UserType.OWNER)
    @GetMapping("/stores")
    public ResponseEntity<ApiResult<List<GetStoreRespDto>>> getAllStoresByOwnerId(@CurrentUser TokenSubject tokenSubject) {
        Long ownerId = tokenSubject.getId();
        return new ResponseEntity<>(ApiResult.success(storeService.getAllStoresByOwnerId(ownerId)), HttpStatus.CREATED);
    }
```
조회하고 있는데 HttpStatus가 Created로 나가고 있습니다!

TIP-Santa committed:
#30 refactor (GetStoreRespDto) : @NoArgsConstructor 추가

TIP-Santa committed:
#30 refactor (StoreController) : HttpStatus.OK 변경

TIP-Santa committed:
chore(DummyObject) : 오류 주석처리
